### PR TITLE
Update Fastmri reader for inference and add support of Tensor for switch_endianness

### LINKED ISSUE #5552: FastMRI reader outputs Tensor which is not supported by Imageloader
**Describe the bug**
FastMRI reader outputs Tensor which is not supported by Imageloader

**To Reproduce**
2. Install '....'
3. Run commands '....'

**Expected behavior**
[switch_endianness](https://github.com/binliunls/MONAI/blob/736053b49f837e382335df461155f407a2cd6ac5/monai/transforms/io/array.py#L271) function reports the bug, due to unsupport of torch.Tensor output by FastMRI reader. 

We should either let FastMRI reader to output numpy array, or let switch_endianness support Tensor.

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes #5552 .

### Description

A few sentences describing the changes proposed in this pull request.
Update Fastmri reader for inference

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @Can-Zhao ,

Is this a blocker issue for MONAI 1.0.1 users to run the MRI reconstruction tutorial and bundle?
If so, is there any workaround to adjust the tutorial and bundle? Because we are building other projects based on the MONAI 1.0.1.
CC @wyli 

Thanks in advance.

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao ,
> 
> Is this a blocker issue for MONAI 1.0.1 users to run the MRI reconstruction tutorial? If so, is there any workaround to adjust the tutorial? Because we are building other projects based on the MONAI 1.0.1. CC @wyli
> 
> Thanks in advance.

It is actually not a blocker for the current tutorial. 
This bug will produce issue during tutorial inference. Yet the current tutorial only contains training and validation. So it does not affect the current tutorial. But in the future, we will need to add inference in tutorial, and it will produce a bug.

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao ,
> 
> Is this a blocker issue for MONAI 1.0.1 users to run the MRI reconstruction tutorial? If so, is there any workaround to adjust the tutorial? Because we are building other projects based on the MONAI 1.0.1. CC @wyli
> 
> Thanks in advance.

Oh, the tutorial actually has inference, as a notebook. https://github.com/Project-MONAI/tutorials/blob/main/reconstruction/MRI_reconstruction/unet_demo/inference.ipynb
Yet the inference data it used has ground truth, so it did not trigger this bug

**Comment by binliunls** [comment]
Hi @Can-Zhao ,
      I think the [switch_endianness](https://github.com/binliunls/MONAI/blob/736053b49f837e382335df461155f407a2cd6ac5/monai/transforms/io/array.py#L271) function reports the bug, due to unsupport of `torch.Tensor`. Should we add a support for `Tensor` type in this function? cc @wyli @Nic-Ma 

Thanks
Bin

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao , I think the [switch_endianness](https://github.com/binliunls/MONAI/blob/736053b49f837e382335df461155f407a2cd6ac5/monai/transforms/io/array.py#L271) function reports the bug, due to unsupport of `torch.Tensor`. Should we add a support for `Tensor` type in this function? cc @wyli @Nic-Ma
> 
> Thanks Bin

Thank you Bin! Yes, I also noticed it, so I changed the data type to numpy array.
I think it is a good idea to add support of Tensor for switch_endianness().

**Comment by wyli** [comment]
> Hi @Can-Zhao , I think the [switch_endianness](https://github.com/binliunls/MONAI/blob/736053b49f837e382335df461155f407a2cd6ac5/monai/transforms/io/array.py#L271) function reports the bug, due to unsupport of `torch.Tensor`. Should we add a support for `Tensor` type in this function? cc @wyli @Nic-Ma
> 
> Thanks Bin

yes, could you please try modify that function, it may or may not work but worthing trying to understand more.

**Comment by Nic-Ma** [comment]
Hi @Can-Zhao ,

How is going on with this PR?

Thanks.

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao ,
> 
> How is going on with this PR?
> 
> Thanks.

Just updated. Could you help look at it? Thank you!

File: monai/transforms/io/array.py [hunk 1]
```diff
@@ -68,6 +68,14 @@ def switch_endianness(data, new="<"):
         data: input to be converted.
         new: the target endianness, currently support "<" or ">".
     """
+    if isinstance(data, torch.Tensor):
+        device = data.device
+        requires_grad: bool = data.requires_grad
+        return (
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/transforms/io/array.py]: Thank you! Done

**Pull request merged**